src/libraries/System.Private.CoreLib/src/System/AppContext.AnyOS.cs

```diff
@@ -11,7 +11,13 @@ public static partial class AppContext
         private static string GetBaseDirectoryCore()
         {
             // Fallback path for hosts that do not set APP_CONTEXT_BASE_DIRECTORY explicitly
-            string? directory = Path.GetDirectoryName(Assembly.GetEntryAssembly()?.Location);
+#if CORERT
+            string? path = Environment.ProcessPath;
```

Comment:
Why is this CORERT only?

Reply:
It would cause different behavior for hosts that don't specify `APP_CONTEXT_BASE_DIRECTORY`: it would return path to the host instead of path to the entrypoint module - this codepath is a fallback for those cases.

CoreRT doesn't set APP_CONTEXT_BASE_DIRECTORY and `Assembly.GetEntryAssembly()?.Location` is not usable in CoreRT because CoreRT follows the .NET 5 single-file behavior and reports an empty string from this API.

Reply:
@CoffeeFlux do we need to do same for mobile?

Reply:
Yes, I do expect we may need this in other situations too. I am not sure what those situations are. It is why I have put it under ifdef for now, to avoid accidentally changing CoreCLR/Mono behavior in subtle way. Maybe the ProcessPath should be part of the fallback path everywhere.

Reply:
I think we should probably incorporate a fallback that isn't dependent on Assembly.Location everywhere; this seems applicable to wasm too. With that said, I'm not sure this particular fallback is appropriate on Android...

Reply:
>  this seems applicable to wasm too

Wasm/Browser has its own implementation of this method that just returns `/`.